src/SpotifyApi.NetCore/SpotifyWebApi.cs

```diff
@@ -173,10 +179,13 @@ private async Task<HttpResponseMessage> PostOrPut(string verb, Uri uri, object d
             switch (verb)
             {
                 case "PUT":
-                    response = await _http.PutAsync(uri, content);
+                    response = await _http.PutAsync(uri, data == null ? null : content);
```

Comment:
Don't change this behaviour - `content` is the correct argument value. Your refactoring tool may be adding unnecessary null checks here?

Reply:
Hi Daniel,

Experimenting I found that my code for PUT it has to be this way or it fails with:

 FollowArtists_ArtistIds_IsTrue
   Source: FollowApiTests.cs line 75
   Duration: 200 ms

  Message: 
    Test method SpotifyApi.NetCore.Tests.FollowApiTests.FollowArtists_ArtistIds_IsTrue threw exception: 
    SpotifyApi.NetCore.SpotifyApiErrorException: Invalid JSON body
  Stack Trace: 
    RestHttpClient.CheckForErrors(HttpResponseMessage response) line 119
    SpotifyWebApi.PostOrPutOrDelete(String verb, Uri uri, Object data, String accessToken) line 196
    SpotifyWebApi.Put(Uri uri, Object data, String accessToken) line 154
    FollowApi.FollowArtists(String[] artistIds, String accessToken) line 162
    FollowApiTests.FollowArtists_ArtistIds_IsTrue() line 85

If you can correct my code and explain what I am doing wrong then would be nice. Otherwise I have to give up on doing all PUT's and POST's because it does not work as it is.

Thanks for your time and patience.

Best Regards,

Akshay Srinivasan

Reply:
OK I will take a look

Reply:
Hmmn, it looks like this is handled inconsistently by the Spotify API. Leave it as it is for now @akshays2112 and I will do some more investigation and fix it after the merge 👍